File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/business_profile/BusinessProfileSampleData.kt
@file:Suppress("PackageNaming")

package app.pantopus.android.ui.screens.business_profile

import app.pantopus.android.ui.screens.saved_places.PendingSavePlace
import app.pantopus.android.ui.theme.PantopusIcon

/**
 * A10.6 — the two hand-authored design frames used by previews, the
 * Paparazzi snapshots, and iOS parity: [populated] (Marlow & Co.
 * Cleaning — open, verified) and [newlyClaimed] (Tide Pool Pet Care —
 * newly-claimed + closed, with `EmptyBlock`s for the unfilled sections).
 *
 * Design reference: `docs/designs/A10/business-frames.jsx`.
 */
object BusinessProfileSampleData {
    val populated: BusinessProfileContent =
        BusinessProfileContent(
            businessId = "marlow",
            header =
                BusinessProfileHeader(
                    displayName = "Marlow & Co. Cleaning",
                    handle = "marlowco",
                    locality = "Elm Park",
                    isVerified = true,
                    logoIcon = PantopusIcon.Sparkles,
                ),
            stats =
                listOf(
                    BusinessStatCell("rating", "4.9", "128 reviews", leadingStar = true, tint = BusinessStatTint.Star),
                    BusinessStatCell("jobs", "340", "Jobs done"),
                    BusinessStatCell("response", "~20m", "Response"),
                ),
            categories =
                listOf(
                    BusinessCategoryChip("cleaning", "Cleaning", PantopusIcon.Sparkles, BusinessCategoryAccent.Cleaning),
                    BusinessCategoryChip("home", "Home & apartment", PantopusIcon.Home, BusinessCategoryAccent.Neutral),
                    BusinessCategoryChip("moveout", "Move-out", PantopusIcon.Package, BusinessCategoryAccent.Neutral),
                    BusinessCategoryChip("eco", "Eco products", PantopusIcon.Leaf, BusinessCategoryAccent.Neutral),
                ),
            about =
                "Family-run cleaning crew that's worked Elm Park homes since 2019. " +
                    "Two-person teams, your own checklist, same crew each visit. We bring " +
                    "eco-safe supplies — you don't stock a thing. Bonded and insured.",
            aboutChips =
                listOf(
                    BusinessAboutChip("bonded", "Bonded & insured", PantopusIcon.Shield),
                    BusinessAboutChip("team", "3 team members", PantopusIcon.Users),
                    BusinessAboutChip("since", "Since 2019", PantopusIcon.CalendarCheck),
                ),
            status =
                BusinessOpenState(
                    isOpen = true,
                    statusLabel = "Open now",
                    statusDetail = "Closes 6:00 PM",
                    chipLabel = "Open now",
                ),
            hours =
                listOf(
                    BusinessHoursRow("mon", "Monday", "8:00 AM – 6:00 PM", isClosed = false, isToday = true),
                    BusinessHoursRow("tue", "Tuesday", "8:00 AM – 6:00 PM", isClosed = false),
                    BusinessHoursRow("wed", "Wednesday", "8:00 AM – 6:00 PM", isClosed = false),
                    BusinessHoursRow("thu", "Thursday", "8:00 AM – 6:00 PM", isClosed = false),
                    BusinessHoursRow("fri", "Friday", "8:00 AM – 5:00 PM", isClosed = false),
                    BusinessHoursRow("sat", "Saturday", "9:00 AM – 2:00 PM", isClosed = false),
                    BusinessHoursRow("sun", "Sunday", "Closed", isClosed = true),
                ),
            serviceArea =
                BusinessServiceArea(
                    title = "Based near 5th & Birch",
                    detail = "Exact address shared after booking",
                    serviceArea = "Serves Elm Park & Cedar Heights — within 4 mi",
                    latitude = 42.37,
                    longitude = -71.11,
                ),
            services =
                listOf(
                    BusinessServiceRow(
                        id = "standard",
                        name = "Standard clean",
                        detail = "2 hr · 2-person team",
                        priceLabel = "from $90",
                        unit = "per visit",
                        icon = PantopusIcon.Droplets,
                    ),
                    BusinessServiceRow(
                        id = "deep",
                        name = "Deep clean",
                        detail = "4 hr · baseboards, inside oven",
                        priceLabel = "from $180",
                        unit = "per visit",
                        icon = PantopusIcon.Sparkles,
                    ),
                    BusinessServiceRow(
                        id = "moveout",
                        name = "Move-out clean",
                        detail = "Empty home · deposit-ready",
                        priceLabel = "from $240",
                        unit = "flat",
                        icon = PantopusIcon.Package,
                    ),
                ),
            gallery =
                listOf(
                    BusinessGalleryItem("kitchen", "Kitchen", BusinessGalleryTint.Primary),
                    BusinessGalleryItem("bath", "Bathroom", BusinessGalleryTint.Success),
                    BusinessGalleryItem("living", "Living room", BusinessGalleryTint.Slate),
                    BusinessGalleryItem("more", null, BusinessGalleryTint.Deep, moreCount = 9),
                ),
            reviewSummary =
                BusinessReviewSummary(
                    average = 4.9,
                    count = 128,
                    distribution = listOf(0.92, 0.06, 0.02, 0.0, 0.0),
                ),
            reviews =
                listOf(
                    BusinessReviewCard(
                        id = "jt",
                        reviewerName = "Jamal T.",
                        reviewerAvatarUrl = null,
                        rating = 5,
                        body =
                            "Same two folks every time, which I love. They remember the dog and " +
                                "shut the gate. Place smells like nothing, which is exactly right.",
                        timestamp = "1w · Standard clean",
                        verified = true,
                    ),
                ),
            dock = BusinessActionDock(BusinessActionDock.Secondary.Book, note = null),
            savedPlace =
                PendingSavePlace(
                    label = "Marlow & Co. Cleaning",
                    latitude = 42.37,
                    longitude = -71.11,
                    city = "Elm Park",
                    sourceId = "marlow",
                ),
            isNewlyClaimed = false,
            phoneNumber = "+15555550100",
            websiteUrl = null,
            viewerIsOwner = false,
        )

    val newlyClaimed: BusinessProfileContent =
        BusinessProfileContent(
            businessId = "tidepool",
            header =
                BusinessProfileHeader(
                    displayName = "Tide Pool Pet Care",
                    handle = "tidepoolpets",
                    locality = "Cedar Heights",
                    isVerified = true,
                    logoIcon = PantopusIcon.PawPrint,
                ),
            stats =
                listOf(
                    BusinessStatCell("rating", "—", "No reviews yet", leadingStar = true, tint = BusinessStatTint.Muted),
                    BusinessStatCell("jobs", "0", "Jobs done"),
                    BusinessStatCell("new", "New", "On Pantopus", tint = BusinessStatTint.Business),
                ),
            categories =
                listOf(
                    BusinessCategoryChip("pet", "Pet care", PantopusIcon.PawPrint, BusinessCategoryAccent.Pet),
                    BusinessCategoryChip("dog", "Dog walking", PantopusIcon.Dog, BusinessCategoryAccent.Neutral),
                ),
            about = null,
            aboutChips = emptyList(),
            status =
                BusinessOpenState(
                    isOpen = false,
                    statusLabel = "Closed now",
                    statusDetail = "Opens tomorrow at 8:00 AM",
                    chipLabel = "Closed · opens 8 AM",
                ),
            hours = emptyList(),
            serviceArea = null,
            services =
                listOf(
                    BusinessServiceRow(
                        id = "walk",
                        name = "30-min dog walk",
                        detail = "Solo walk · your route",
                        priceLabel = "$22",
                        unit = "per walk",
                        icon = PantopusIcon.PawPrint,
                    ),
                    BusinessServiceRow(
                        id = "dropin",
                        name = "Drop-in visit",
                        detail = "Feed, water, playtime",
                        priceLabel = "$20",
                        unit = "per visit",
                        icon = PantopusIcon.Home,
                    ),
                ),
            gallery = emptyList(),
            reviewSummary = null,
            reviews = emptyList(),
            dock =
                BusinessActionDock(
                    BusinessActionDock.Secondary.Call,
                    note = "Closed now — messages answered at 8 AM",
                ),
            isNewlyClaimed = true,
            phoneNumber = "+15555550111",
            websiteUrl = null,
            viewerIsOwner = false,
        )
}

File: frontend/apps/ios/Pantopus/Features/BusinessProfile/BusinessProfileSampleData.swift
//
//  BusinessProfileSampleData.swift
//  Pantopus
//
//  A10.6 — the two hand-authored design frames used by previews, the
//  snapshot reference, and Android parity: `FrameBizPopulated`
//  (Marlow & Co. Cleaning — open, verified, full profile) and
//  `FrameBizNew` (Tide Pool Pet Care — newly-claimed + closed, with
//  `EmptyBlock`s for the unfilled sections).
//
//  Design reference: `docs/designs/A10/business-frames.jsx`.
//

import Foundation

enum BusinessProfileSampleData {
    /// FrameBizPopulated — Marlow & Co. Cleaning (open, verified).
    static let populated = BusinessProfileContent(
        businessId: "marlow",
        header: BusinessProfileHeader(
            displayName: "Marlow & Co. Cleaning",
            handle: "marlowco",
            locality: "Elm Park",
            isVerified: true,
            logoIcon: .sparkles
        ),
        stats: [
            BusinessStatCell(id: "rating", value: "4.9", label: "128 reviews", leadingStar: true, tint: .star),
            BusinessStatCell(id: "jobs", value: "340", label: "Jobs done"),
            BusinessStatCell(id: "response", value: "~20m", label: "Response")
        ],
        categories: [
            BusinessCategoryChip(id: "cleaning", label: "Cleaning", icon: .sparkles, accent: .cleaning),
            BusinessCategoryChip(id: "home", label: "Home & apartment", icon: .home, accent: .neutral),
            BusinessCategoryChip(id: "moveout", label: "Move-out", icon: .package, accent: .neutral),
            BusinessCategoryChip(id: "eco", label: "Eco products", icon: .leaf, accent: .neutral)
        ],
        about: "Family-run cleaning crew that's worked Elm Park homes since 2019. "
            + "Two-person teams, your own checklist, same crew each visit. We bring "
            + "eco-safe supplies — you don't stock a thing. Bonded and insured.",
        aboutChips: [
            BusinessAboutChip(id: "bonded", label: "Bonded & insured", icon: .shield),
            BusinessAboutChip(id: "team", label: "3 team members", icon: .users),
            BusinessAboutChip(id: "since", label: "Since 2019", icon: .calendarCheck)
        ],
        status: BusinessOpenState(
            isOpen: true,
            statusLabel: "Open now",
            statusDetail: "Closes 6:00 PM",
            chipLabel: "Open now"
        ),
        hours: [
            BusinessHoursRow(id: "mon", dayLabel: "Monday", timeLabel: "8:00 AM – 6:00 PM", isClosed: false, isToday: true),
            BusinessHoursRow(id: "tue", dayLabel: "Tuesday", timeLabel: "8:00 AM – 6:00 PM", isClosed: false),
            BusinessHoursRow(id: "wed", dayLabel: "Wednesday", timeLabel: "8:00 AM – 6:00 PM", isClosed: false),
            BusinessHoursRow(id: "thu", dayLabel: "Thursday", timeLabel: "8:00 AM – 6:00 PM", isClosed: false),
            BusinessHoursRow(id: "fri", dayLabel: "Friday", timeLabel: "8:00 AM – 5:00 PM", isClosed: false),
            BusinessHoursRow(id: "sat", dayLabel: "Saturday", timeLabel: "9:00 AM – 2:00 PM", isClosed: false),
            BusinessHoursRow(id: "sun", dayLabel: "Sunday", timeLabel: "Closed", isClosed: true)
        ],
        serviceArea: BusinessServiceArea(
            title: "Based near 5th & Birch",
            detail: "Exact address shared after booking",
            serviceArea: "Serves Elm Park & Cedar Heights — within 4 mi",
            latitude: 42.37,
            longitude: -71.11
        ),
        services: [
            BusinessServiceRow(
                id: "standard",
                name: "Standard clean",
                detail: "2 hr · 2-person team",
                priceLabel: "from $90",
                unit: "per visit",
                icon: .droplets
            ),
            BusinessServiceRow(
                id: "deep",
                name: "Deep clean",
                detail: "4 hr · baseboards, inside oven",
                priceLabel: "from $180",
                unit: "per visit",
                icon: .sparkles
            ),
            BusinessServiceRow(
                id: "moveout",
                name: "Move-out clean",
                detail: "Empty home · deposit-ready",
                priceLabel: "from $240",
                unit: "flat",
                icon: .package
            )
        ],
        gallery: [
            BusinessGalleryItem(id: "kitchen", label: "Kitchen", tint: .primary),
            BusinessGalleryItem(id: "bath", label: "Bathroom", tint: .success),
            BusinessGalleryItem(id: "living", label: "Living room", tint: .slate),
            BusinessGalleryItem(id: "more", label: nil, tint: .deep, moreCount: 9)
        ],
        reviewSummary: BusinessReviewSummary(
            average: 4.9,
            count: 128,
            distribution: [0.92, 0.06, 0.02, 0, 0]
        ),
        reviews: [
            BusinessReviewCard(
                id: "jt",
                reviewerName: "Jamal T.",
                reviewerAvatarURL: nil,
                rating: 5,
                body: "Same two folks every time, which I love. They remember the dog and shut "
                    + "the gate. Place smells like nothing, which is exactly right.",
                timestamp: "1w · Standard clean",
                verified: true
            )
        ],
        dock: BusinessActionDock(secondary: .book, note: nil),
        savedPlace: PendingSavePlace(
            label: "Marlow & Co. Cleaning",
            latitude: 42.37,
            longitude: -71.11,
            city: "Elm Park",
            sourceId: "marlow"
        ),
        isNewlyClaimed: false,
        phoneNumber: "+15555550100",
        websiteURL: nil,
        viewerIsOwner: false
    )

    /// FrameBizNew — Tide Pool Pet Care (newly-claimed + closed).
    /// About / Hours / Service area / Recent work / Reviews are all empty
    /// so the secondary frame demonstrates the `EmptyBlock` states.
    static let newlyClaimed = BusinessProfileContent(
        businessId: "tidepool",
        header: BusinessProfileHeader(
            displayName: "Tide Pool Pet Care",
            handle: "tidepoolpets",
            locality: "Cedar Heights",
            isVerified: true,
            logoIcon: .pawPrint
        ),
        stats: [
            BusinessStatCell(id: "rating", value: "—", label: "No reviews yet", leadingStar: true, tint: .muted),
            BusinessStatCell(id: "jobs", value: "0", label: "Jobs done"),
            BusinessStatCell(id: "new", value: "New", label: "On Pantopus", tint: .business)
        ],
        categories: [
            BusinessCategoryChip(id: "pet", label: "Pet care", icon: .pawPrint, accent: .pet),
            BusinessCategoryChip(id: "dog", label: "Dog walking", icon: .dog, accent: .neutral)
        ],
        about: nil,
        aboutChips: [],
        status: BusinessOpenState(
            isOpen: false,
            statusLabel: "Closed now",
            statusDetail: "Opens tomorrow at 8:00 AM",
            chipLabel: "Closed · opens 8 AM"
        ),
        hours: [],
        serviceArea: nil,
        services: [
            BusinessServiceRow(
                id: "walk",
                name: "30-min dog walk",
                detail: "Solo walk · your route",
                priceLabel: "$22",
                unit: "per walk",
                icon: .pawPrint
            ),
            BusinessServiceRow(
                id: "dropin",
                name: "Drop-in visit",
                detail: "Feed, water, playtime",
                priceLabel: "$20",
                unit: "per visit",
                icon: .home
            )
        ],
        gallery: [],
        reviewSummary: nil,
        reviews: [],
        dock: BusinessActionDock(
            secondary: .call,
            note: "Closed now — messages answered at 8 AM"
        ),
        isNewlyClaimed: true,
        phoneNumber: "+15555550111",
        websiteURL: nil,
        viewerIsOwner: false
    )
}
